server/routes.go

```diff
@@ -1640,3 +1642,23 @@ func handleScheduleError(c *gin.Context, name string, err error) {
 		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
 	}
 }
+
+var thinkTagRegexp = regexp.MustCompile(`<think>(?s).*?</think>(\n)*`)
+
+func filterThinkTags(msgs []api.Message, model *Model) []api.Message {
+	if model.Config.ModelFamily == "qwen3" || strings.HasPrefix(model.ShortName, "deepseek-r1:") {
```

Comment:
This would also be helpful for:
- https://ollama.com/library/cogito
- https://ollama.com/library/qwq
- https://ollama.com/library/openthinker

Although I don't think there's an easy way to check for this, so maybe we don't add too many cases

Reply:
will do a follow up soon to add more, want to test them a bit first